gcsweb/cmd/gcsweb/gcsweb.go

```diff
@@ -67,26 +67,94 @@ func (ss *strslice) Set(value string) error {
 	return nil
 }
 
-// Only buckets in this list will be served.
-var allowedBuckets strslice
+type options struct {
+	flPort int
+
+	flIcons        string
+	flStyles       string
+	oauthTokenFile string
+
+	flVersion bool
+
+	// Only buckets in this list will be served.
+	allowedBuckets strslice
+}
+
+var flUpgradeProxiedHTTPtoHTTPS bool
```

Comment:
I left this as global like before, because there are many funcs that use this variable in some checks, and I didn't want to turn them all to methods and pass this variable around.